### PR TITLE
Add the possibility to retrieve the current user groups

### PR DESCRIPTION
**Pull request by fviale**
- groups are available in a UserData class containing also the user name.
- Added getCurrentUserData method in Scheduler and ResourceManager interfaces
- added logins/sessionid/{sessionId}/userdata rest call in Scheduler and RM rest interfaces
- added currentuserdata rest call in studio interface
- added tests in AuthenticationTest and SchedulerClientTest which checks that the UserData can be retrieved.

### CODE REVIEW

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 616 issues
* MAJOR 2 major
* MINOR 614 minor

#### Top 10 issues

1. MAJOR [RMRest.java#L221](https://github.com/ow2-proactive/scheduling/blob/74eb5275abe0d94a245849dc5772d1f09d88d1bd/rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java#L221): Either log or rethrow this exception. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [SchedulerStateRest.java#L2991](https://github.com/ow2-proactive/scheduling/blob/74eb5275abe0d94a245849dc5772d1f09d88d1bd/rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java#L2991): Either log or rethrow this exception. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. MINOR [UserData.java#L44](https://github.com/ow2-proactive/scheduling/blob/74eb5275abe0d94a245849dc5772d1f09d88d1bd/common/common-api/src/main/java/org/ow2/proactive/authentication/UserData.java#L44): Move this variable to comply with Java Code Conventions. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1213)
1. MINOR [UserData.java#L46](https://github.com/ow2-proactive/scheduling/blob/74eb5275abe0d94a245849dc5772d1f09d88d1bd/common/common-api/src/main/java/org/ow2/proactive/authentication/UserData.java#L46): Move this variable to comply with Java Code Conventions. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1213)
1. MINOR [Scheduler.java#L83](https://github.com/ow2-proactive/scheduling/blob/74eb5275abe0d94a245849dc5772d1f09d88d1bd/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L83): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [Scheduler.java#L92](https://github.com/ow2-proactive/scheduling/blob/74eb5275abe0d94a245849dc5772d1f09d88d1bd/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L92): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [Scheduler.java#L114](https://github.com/ow2-proactive/scheduling/blob/74eb5275abe0d94a245849dc5772d1f09d88d1bd/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L114): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [Scheduler.java#L157](https://github.com/ow2-proactive/scheduling/blob/74eb5275abe0d94a245849dc5772d1f09d88d1bd/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L157): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [Scheduler.java#L187](https://github.com/ow2-proactive/scheduling/blob/74eb5275abe0d94a245849dc5772d1f09d88d1bd/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L187): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [Scheduler.java#L231](https://github.com/ow2-proactive/scheduling/blob/74eb5275abe0d94a245849dc5772d1f09d88d1bd/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L231): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java [hunk 1]
```diff
@@ -2975,6 +2976,27 @@ public String getLoginFromSessionId(@PathParam("sessionId") String sessionId) {
...
+    @Path("logins/sessionid/{sessionId}/userdata")
+    @Produces("application/json")
+    public UserData getUserDataFromSessionId(@PathParam("sessionId") String sessionId) {
+        if (sessionId != null && sessionStore.exists(sessionId)) {
+            try {
+                Scheduler scheduler = sessionStore.get(sessionId).getScheduler();
+                UserData userData = scheduler.getCurrentUserData();
+                return userData;
+            } catch (NotConnectedException e) {
+                return null;
```
Comment (tobwiens) [inline rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java]: Can we return here http code  403

File: common/common-api/src/main/java/org/ow2/proactive/authentication/UserData.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Set;
+
+
+/**
+ * A class representing a user information
+ *
+ * @author ActiveEon Team
+ * @since 28/07/2017
+ */
+public class UserData implements Serializable {
```
Comment (tobwiens) [inline common/common-api/src/main/java/org/ow2/proactive/authentication/UserData.java]: What if you make this class immutable with a builder. Lombok can do the job. Does that work in this case?

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 616 issues
* MAJOR 2 major
* MINOR 614 minor

#### Top 10 issues

1. MAJOR [RMRest.java#L221](https://github.com/ow2-proactive/scheduling/blob/99953ee4d8aec531a68c00498a85dab8d7857b60/rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java#L221): Either log or rethrow this exception. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [SchedulerStateRest.java#L2991](https://github.com/ow2-proactive/scheduling/blob/99953ee4d8aec531a68c00498a85dab8d7857b60/rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java#L2991): Either log or rethrow this exception. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)
1. MINOR [UserData.java#L44](https://github.com/ow2-proactive/scheduling/blob/99953ee4d8aec531a68c00498a85dab8d7857b60/common/common-api/src/main/java/org/ow2/proactive/authentication/UserData.java#L44): Move this variable to comply with Java Code Conventions. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1213)
1. MINOR [UserData.java#L46](https://github.com/ow2-proactive/scheduling/blob/99953ee4d8aec531a68c00498a85dab8d7857b60/common/common-api/src/main/java/org/ow2/proactive/authentication/UserData.java#L46): Move this variable to comply with Java Code Conventions. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1213)
1. MINOR [Scheduler.java#L83](https://github.com/ow2-proactive/scheduling/blob/99953ee4d8aec531a68c00498a85dab8d7857b60/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L83): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [Scheduler.java#L92](https://github.com/ow2-proactive/scheduling/blob/99953ee4d8aec531a68c00498a85dab8d7857b60/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L92): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [Scheduler.java#L114](https://github.com/ow2-proactive/scheduling/blob/99953ee4d8aec531a68c00498a85dab8d7857b60/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L114): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [Scheduler.java#L157](https://github.com/ow2-proactive/scheduling/blob/99953ee4d8aec531a68c00498a85dab8d7857b60/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L157): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [Scheduler.java#L187](https://github.com/ow2-proactive/scheduling/blob/99953ee4d8aec531a68c00498a85dab8d7857b60/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L187): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [Scheduler.java#L231](https://github.com/ow2-proactive/scheduling/blob/99953ee4d8aec531a68c00498a85dab8d7857b60/scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java#L231): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: common/common-api/src/main/java/org/ow2/proactive/authentication/UserData.java (hunk 2 above)
Comment (fviale) [inline common/common-api/src/main/java/org/ow2/proactive/authentication/UserData.java]: It is made to be returned as a rest call, via dozer mapping, so it cannot be immutable

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java (hunk 1 above)
Comment (fviale) [inline rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java]: There is currently no authorization mechanism for the rest API directly, and here we simply check if a session exists. And if it exists, we call the getCurrentUserData() method which has no specific security policy (i.e. an authenticated user can always know his username or his group). 
I don't know well how to check access here.

File: common/common-api/src/main/java/org/ow2/proactive/authentication/UserData.java (hunk 2 above)
Comment (tobwiens) [inline common/common-api/src/main/java/org/ow2/proactive/authentication/UserData.java]: Yeah I thought so but had hope ;)

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java (hunk 1 above)
Comment (tobwiens) [inline rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java]: 👍

**Pull request merged**